### ISSUE TITLE
Workbooks

### ISSUE THREAD

**Issue by nickmcintyre** [JithinKS97/dynamic-learning#39]
I propose adapting our mental model of **Lessonplans** to accommodate instruction, practice, and assessment. After students have viewed a **Lesson** and interacted with a **Simulation**, they should be provided an opportunity to demonstrate their mastery of the concept being taught. The functionality would be more accurately modeled as a **Workbook** than a lesson plan, which is a planning document geared towards teachers and administrators. The transition would simply require renaming to start and would be completed when @ashneeldas2 gets to assessment later this summer.

What do y'all think?

**Comment by JithinKS97** [comment]
Initially,  when I started developing the app,  I did not have the idea of assessment in my mind. The idea was that teachers use the space to screen record or to present the lesson directly to the class room. Thats why I named it as Lessonplan. But I agree,  if we are planning to accomodate assessments in it (Which is very cool !!!) like for instance ( multiple choice question or may be finding out a value by interacting with the sim or a task to redesign the sim ),  the terminology Workbook makes more sense. But do we just need to bring change superficially. Or do we need to change the naming in the code level which would take way more effort.

**Comment by nickmcintyre** [comment]
Yeah, "simply" is a misnomer. We can get started by changing the display name, but I think it's worth changing names throughout the code base. Happy to tackle this one.

**Comment by JithinKS97** [comment]
Are we good to close this issue @nickmcintyre ?

**Comment by nickmcintyre** [comment]
What do y'all think about having a couple of modes (e.g., "Edit" and "View") for workbooks? Some of the right-hand menu options only seem relevant for creating and editing, and others will be needed while students work.

**Comment by JithinKS97** [comment]
In the dashboard,  users already have the provision to see the preview of the workbook (Viewing) and they enter the work book creator (Editing) by clicking the open button. Could you please give a little more detail about the editing and viewing option that you are suggesting?

**Comment by nickmcintyre** [comment]
Thanks for clarifying–the distinction I have in mind is more accurately framed as "Create" and "Complete" (or something along those lines).

I just created a workbook as a teacher and assigned it to a class. Students in the class can find a link to the workbook from their classes page, which takes them to the workbook creator. Instead, it would be helpful to begin tailoring that interaction with the idea that a portion of the workbook is fixed by the teacher (e.g., students might "accidentally" delete the assessment :).

**Comment by JithinKS97** [comment]
Currently, even if a student opens the workbook created by a teacher, he can't edit the workbook since he is not the owner. Though he can fork the workbook to create a copy in which he can have changes which he can save.

I think we must think of ways in which the teacher can track the forks created by the students and also the communication.

Currently when a fork is created, a request forum is created along with it. May be what we can do is for all the forks, the discussion forum can be pointed to the parent workbook. That way the discussion forum is the ground in which teachers and his students have the discussions.

**Comment by nickmcintyre** [comment]
I'm a fan of class-wide discussions for workbooks and helping students to organize around specific topics/problems. Linking to a direct view of each others' workbook could be troublesome depending on the type(s) of assessment used.

**Comment by JithinKS97** [comment]
OK. So may be we can create an interface which lets only the teacher to view the forks of the students (Not the other students). For the class wide discussions, we may use the present discussion forum and may be we can also create an interface where the teacher and the individual student can have discussions about that student's particular fork (Like maybe a chat?)

**Comment by nickmcintyre** [comment]
Direct messaging is definitely useful for individual feedback and questions that students may not want to ask in front of peers.

The current discussion forums are a great foundation to build upon. They do require a little adaptation from their original application, though. For example, I think the "Uploaded sims" column probably isn't relevant for most academic discussions.

**Comment by ashneeldas2** [comment]
What do y’all think of this for workbooks? 

- Allow embedding of questions into workbooks, which students can then answer and receive a grade after completing the workbook. This would be a system different from existing learning management systems, but also useful in that it enforces that the student actually read the slides.
- Have slide templates (similar to how Google Slides does it) which have an area for content, an area for questions, etc.

**Comment by JithinKS97** [comment]
Yeah but before going into grading, we should think of ways for teachers to conduct the assessment (What are the different ways, is it MCQ? Textual input? Checkboxes? ) and storing the responses of the students in the slides.
